src/main/java/services/impl/FormRecordProcessorImpl.java

```diff
@@ -35,11 +41,32 @@
  * <p>
  * Created by wpride1 on 8/20/15.
  */
-public class FormRecordProcessorHelper extends XmlFormRecordProcessor {
+public class FormRecordProcessorImpl extends XmlFormRecordProcessor implements RecordProcessorService {
 
-    private static final Log log = LogFactory.getLog(FormRecordProcessorHelper.class);
+    @Autowired
+    protected StatsDClient datadogStatsDClient;
 
-    public static void processXML(FormplayerTransactionParserFactory factory, String fileText) throws IOException, XmlPullParserException, UnfullfilledRequirementsException, InvalidStructureException {
+    @Autowired
+    protected RestoreFactory restoreFactory;
+
+    private final Log log = LogFactory.getLog(FormRecordProcessorImpl.class);
+
+    // This function will only wipe user DBs when they have expired, otherwise will incremental sync
+    public UserSqlSandbox performSync(RestoreFactory restoreFactory) throws Exception {
```

Comment:
Rather than moving all of the things that you want to time into the same parent class, could you add a `@ReportTiming` annotation with an aspect that does times it and sends to datadog? Or was there a different reason for reorganizing these functions into a single parent class?